src/ParallelAlgorithms/Interpolation/Protocols/InterpolationTargetTag.hpp

```diff
@@ -0,0 +1,123 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+#include <type_traits>
+
+#include "ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/ComputeTargetPoints.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/ComputeVarsToInterpolate.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/PostInterpolationCallback.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+
+namespace intrp::protocols {
+namespace detail {
+// Both will have `::value = true` because we want the static assert in the
+// protocol to pass regardless of if the `compute_vars_to_interpolate` alias
+// exists if the conforming type. It's just that if the alias does exist,
+// there's an additional static assert to ensure that the alias conforms to the
+// ComputeVarsToInterpolate protocol
+template <typename ConformingType, bool HasAlias>
+struct if_alias_exists_assert_conforms_to;
+
+template <typename ConformingType>
+struct if_alias_exists_assert_conforms_to<ConformingType, false> {
+  static constexpr bool value = true;
+};
+
+template <typename ConformingType>
+struct if_alias_exists_assert_conforms_to<ConformingType, true> {
+  static constexpr bool value = true;
+  static_assert(tt::assert_conforms_to<
+                typename ConformingType::compute_vars_to_interpolate,
+                ComputeVarsToInterpolate>);
+};
+
+template <typename ConformingType, bool HasAlias>
+constexpr bool if_alias_exists_assert_conforms_to_v =
+    if_alias_exists_assert_conforms_to<ConformingType, HasAlias>::value;
+}  // namespace detail
+
+/*!
+ * \brief A protocol for `InterpolationTargetTag`s that are used in the
+ * intrp::InterpolationTarget parallel component.
+ *
+ * \details A struct conforming to the `InterpolationTarget` protocol must
+ * have
+ *
+ * - a type alias `temporal_id` that tells the interpolation target what values
+ *   of time to use.
+ *
+ * - a static function `name()` that returns a `std::string` used to uniquely
+ *   identify this InterpolationTargetTag
+ *
+ * - a type alias `vars_to_interpolate_to_target` which is a `tmpl::list` of
+ *   tags describing variables to interpolate. Will be used to construct a
+ *   `Variables`.
+ *
+ * - a type alias  `compute_items_on_target` which is a `tmpl::list` of compute
+ *   items that uses `vars_to_interpolate_to_target` as input.
+ *
+ * - a type alias `compute_target_points` that conforms to the
+ *   intrp::protocols::ComputeTargetPoints protocol. See this protocol for
+ *   requirements on `compute_target_points`.
```

Comment:
You can delete this last sentence. Perhaps add a note how the alias is used instead. (same comment for the entries below)